File: src/main/java/org/taktik/icure/logic/impl/ApplicationSettingsLogicImpl.java
package org.taktik.icure.logic.impl;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.taktik.icure.dao.ApplicationSettingsDAO;
import org.taktik.icure.entities.ApplicationSettings;
import org.taktik.icure.logic.ApplicationSettingsLogic;

@Service
public class ApplicationSettingsLogicImpl extends GenericLogicImpl<ApplicationSettings, ApplicationSettingsDAO> implements ApplicationSettingsLogic {

    private ApplicationSettingsDAO applicationSettingsDAO;

    @Override
    protected ApplicationSettingsDAO getGenericDAO() {
        return this.applicationSettingsDAO;
    }

    @Autowired
    public void setApplicationSettingsDAO(ApplicationSettingsDAO dao) { this.applicationSettingsDAO = dao; }

    @Override
    public ApplicationSettings modifyApplicationSettings(ApplicationSettings applicationSettings) {
        return applicationSettingsDAO.save(applicationSettings);
    }

    @Override
    public ApplicationSettings createApplicationSettings(ApplicationSettings applicationSettings) {
        return applicationSettingsDAO.create(applicationSettings);
    }
}

File: src/main/kotlin/org/taktik/icure/asynclogic/impl/ApplicationSettingsLogicImpl.kt
package org.taktik.icure.asynclogic.impl

import org.springframework.stereotype.Service
import org.taktik.icure.asyncdao.ApplicationSettingsDAO
import org.taktik.icure.asynclogic.ApplicationSettingsLogic
import org.taktik.icure.asynclogic.AsyncSessionLogic
import org.taktik.icure.entities.ApplicationSettings

@Service
class ApplicationSettingsLogicImpl(private val applicationSettingsDAO: ApplicationSettingsDAO, sessionLogic: AsyncSessionLogic) : GenericLogicImpl<ApplicationSettings, ApplicationSettingsDAO>(sessionLogic), ApplicationSettingsLogic {
    override fun getGenericDAO(): ApplicationSettingsDAO {
        return applicationSettingsDAO
    }
}
